pkg/dynamic/client.go

```diff
@@ -0,0 +1,89 @@
+package dynamic
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	kubeerror "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+type ApiHelper struct {
+	Discover discovery.DiscoveryInterface
+	Client   dynamic.Interface
+}
+
+func NewApiHelper(c *rest.Config) (*ApiHelper, error) {
+	client, err := dynamic.NewForConfig(c)
+	if err != nil {
+		return nil, fmt.Errorf("could not create dynamic client: %v", err)
+	}
+	discover, err := discovery.NewDiscoveryClientForConfig(c)
+	if err != nil {
+		return nil, fmt.Errorf("could not create discovery client: %v", err)
+	}
+	return &ApiHelper{
+		Discover: discover,
+		Client:   client,
+	}, nil
+}
+
+func (a *ApiHelper) CreateObject(obj *unstructured.Unstructured) error {
+	groupResources, err := restmapper.GetAPIGroupResources(a.Discover)
```

Comment:
Just based on the flow of data in this function you could probably reduce this into a set of private or public functions and separate out some of the behavior into sub-components + with tests.  

Did you copy some of this from upstream examples in the latest client-go?

Reply:
I couldn't find any examples of the dynamic client upstream. this was the result of reading a lot of code fairly quickly